src/test/librados_test_stub/LibradosTestStub.cc

```diff
@@ -1366,6 +1373,13 @@ int cls_cxx_write_full(cls_method_context_t hctx, bufferlist *inbl) {
   return ctx->io_ctx_impl->write_full(ctx->oid, *inbl, ctx->snapc);
 }
 
+int cls_cxx_replace(cls_method_context_t hctx, int ofs, int len,
+                    bufferlist *inbl) {
+  librados::TestClassHandler::MethodContext *ctx =
+    reinterpret_cast<librados::TestClassHandler::MethodContext*>(hctx);
+  return ctx->io_ctx_impl->replace(ctx->oid, *inbl, len, ofs, ctx->snapc);
```

Comment:
Nit: why not just issue a truncate and write instead of adding a new op?

Reply:
updated. Thanks.